packages/web/docs/src/components/company-testimonials.tsx

```diff
@@ -95,37 +95,74 @@ const testimonials: Testimonial[] = [
 ];
 
 export function CompanyTestimonialsSection({ className }: { className?: string }) {
+  const tabsListRef = useRef<HTMLDivElement>(null);
+  const scrollviewRef = React.useRef<HTMLDivElement>(null);
+  const updateDotsOnScroll = useRef<(event: React.UIEvent) => void>(null);
+  updateDotsOnScroll.current ||= debounce(() => {
+    const scrollview = scrollviewRef.current;
+    const tabsList = tabsListRef.current;
+    if (!scrollview || !tabsList) return;
+    const scrollLeft = scrollview.scrollLeft;
+    const scrollWidth = scrollview.scrollWidth;
+    const index = Math.round((scrollLeft / scrollWidth) * testimonials.length);
+
+    const tabs = tabsList.querySelectorAll('[role="tab"]');
+    for (const [i, tab] of tabs.entries()) {
+      tab.setAttribute('data-state', i === index ? 'active' : 'inactive');
+    }
```

Comment:
It feels kinda primitive, but the scroll-snap event is still in the working group, so I'm doing this to sync scrolls to changes in Radix data-state.